ui-components/autocompleter.md

```diff
@@ -26,21 +26,134 @@ The two arguments this method take are:
+> **Note**: If two or more autocompleters use the same trigger character, then the fetched results will be merged together before being displayed.
 
-The `fetch` results should be a list of objects with the following details:
-* `value`: Value of the item. This will be passed to the `onAction` callback when selected.
-* `text`: Text to display for the item.
-* `icon`: Name of the icon to be displayed. Must be a single unicode character or correspond to an icon in the icon pack.
+### Usage of fetch
 
-> **Note**: If two or more autocompleters use the same trigger character, then the fetched results will be merged together before being displayed.
+The `fetch` function is called when the trigger `char` is pressed and the `matches` predicate returns `true`. The `fetch` function passed the matched text pattern and returns a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) containing matching results, specified as either [AutocompleteItems](#autocompleteitem) or [CardMenuItems](#cardmenuitem). This allows for asynchronous fetching of the items. The `fetchOptions` passed to the `fetch` function is by default an empty object, however using the [reload](#api) API, additional options can be passed to fetch a different set of results.
+
+There are two types of items:
+
+* [AutocompleteItem](#autocompleteitem)
+* [CardMenuItem](#cardmenuitem)
+
+#### AutocompleteItem
+
+This is the standard item for the autocompleter. If no type is specified, autocompeter items are assumed to be of type `AutocompleteItem`.
+
+| Name | Value | Requirement | Description |
+| ---- | ----- | ----------- | ----------- |
+| value | string | optional | Value of the item. This will be passed to the `onAction` callback when selected. |
+| text | string | optional | Text to display for the item. |
+| icon | string | optional | Name of the icon to be displayed. Must be a either single unicode character or an icon from: the [icon pack]({{site.baseurl}}/advanced/editor-icon-identifiers/), a [custom icon pack]({{site.baseurl}}/advanced/creating-an-icon-pack/), or an icon added using the [`addIcon` API]({{site.baseurl}}/api/tinymce.editor.ui/tinymce.editor.ui.registry/#addicon). |
+
+```js
+{
+  type: 'autocompleteitem',
+  value: 'John Doe',
+  text: 'John Doe',
+  icon: 'my_icon'
+}
+```
+
+{% comment %} TODO: DOC-719 and TINY-6593: Move CardMenuItem/subitems specification to menuitems.md {% endcomment %}
+
+#### CardMenuItem
+
+{{site.requires_5_6v}}
+
+The `CardMenuItem` allow for customization of layout and content. This is done by constructing an array of subcomponent specifications which will decide the structure of the final item. A classical example of this would be the need to display a "user" item containing a profile picture, a user name, and a description.
```

Comment:
```suggestion
The `CardMenuItem` allows customization of layout and content. This is done by constructing an array of subcomponent specifications which will determine the structure of the final item. A classical example of this would be the need to display a "user" item containing a profile picture, a user name, and a description.
```